### PR TITLE
🗿🗿🗿

### PR DESCRIPTION
**Pull request by MeguminSama**
🗿

### CODE REVIEW

**Comment by wingio** [comment]
🗿🗿🗿

**Comment by MeguminSama** [comment]
🗿🗿🗿

**Comment by GlitchChan** [comment]
🗿🗿🗿

**Comment by ActuallyTheSun** [comment]
🗿🗿🗿

**Comment by ActuallyTheSun** [comment]
🗿🗿🗿

**Comment by Vendicated** [comment]
🗿 🗿 🗿

**Comment by Iucism** [comment]
🗿 🗿 🗿

**Comment by ItzOnlyAnimal** [comment]
🗿🗿🗿

**Comment by MeguminSama** [comment]
🗿🗿🗿

File: src/plugins/moyai.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    description: "🗿🗿🗿🗿🗿🗿🗿🗿",
+    execute: async (event: IMessageCreate) => {
+        if (event?.type !== "MESSAGE_CREATE") return;
+        if (!event.message?.content) return;
+        if (event.message.state === "SENDING") return;
+        if (event.optimistic) return;
+
+        const isInGuildChannel =
+            window.location.pathname.startsWith("/channels/");
+        if (!isInGuildChannel) return;
```
Comment (Vendicated) [inline src/plugins/moyai.ts:28]: no dms? 🗿

File: src/plugins/moyai.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        for (let i = 0; i < moyaiCount; i++) {
+            const audioElement = document.createElement("audio");
+            audioElement.src = MOYAI_URL;
+            audioElement.play();
+            await new Promise(resolve => setTimeout(resolve, 300));
+        }
+    },
+    patches: [
+        {
+            find: "MESSAGE_CREATE:function(",
```
Comment (Vendicated) [inline src/plugins/moyai.ts:45]: i'd just FluxDispatcher.subscribe("MESSAGE_CREATE", )

**Review by Vendicated — CHANGES_REQUESTED** [review]

File: src/plugins/moyai.ts (hunk 1 above)
Comment (MeguminSama) [reply src/plugins/moyai.ts:28]: 🗿🗿🗿🗿🗿🗿🗿🗿

File: src/plugins/moyai.ts (hunk 2 above)
Comment (MeguminSama) [reply src/plugins/moyai.ts:45]: 🗿🗿🗿🗿🗿🗿🗿🗿🗿🗿🗿🗿🗿🗿🗿🗿

File: src/plugins/moyai.ts (hunk 1 above)
Comment (MeguminSama) [reply src/plugins/moyai.ts:28]: Works in DMs

Comment (Vendicated) [reply src/plugins/moyai.ts:28]: then it shouldnt be named isInGuildChannel 🗿

File: src/plugins/moyai.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+            // If emoji starts  or ends with (moyai|moai)
+            if (/^(moyai|moai)/i.test(name) || /(moyai|moai)$/i.test(name)) {
+                moyaiCount++;
+            }
+        }
+    }
+
+    // Maximum moyai...
+    if (moyaiCount > 10) moyaiCount = 10;
```
Comment (Vendicated) [inline src/plugins/moyai.ts:79]: return Math.min(moyaiCount, 10)

**Pull request merged**